pkg/mgmt/restore/restore.go

```diff
@@ -0,0 +1,246 @@
+/*
```

Comment:
Can we add a doc.go to this folder that lists the various low-level cases that are supported and any limitations? 

Few things that are not clear by going through the code easily are:
- how the restore differentiate between zvol and zfs datasets?
- how does the file-system information get passed on to the restore process?

For example: 

The restore flow is as follows:
- plugin creates a restore PV (is this correct?)
- plugin create a ZFS restore CR to restore the data from remote backup into the restore PV?
- restore controller (on node) keeps a a watch for new CRs associated with the node id. 
- if status != init or marked for deletion, restore controller will execute the zfs recv | create command. 

Limitations:
- If the restore fails due to network issues? Will it be re-attempted?
- If the restore doesn't have the specified backup, will it marked as failed?
- Similar to the need for nodeID being same, the expectation is also that capacity should be available on the dest node/pool?
- What happens when the same volume is restored twice or if the dest pool already has a volume with the same name.

Reply:
added.